consumer_monitor.go

```diff
@@ -136,13 +136,15 @@ func (m *consumerMonitor) run(ctx context.Context, subscriptionEvents chan *subs
 				log.Errorf("Fetching offsets failed %d times in a row. Consumer status data is stale.",
 					m.scheduler.failureCount)
 
-				if !errors.Is(err, syscall.EPIPE) {
+				if m.connectionResetDisabled {
```

Comment:
If it's configured to reset connections and the error is not a broken pipe, maybe it still needs `continue`, leaving it as `or` condition?

Reply:
The issue is that the broken pipe error was the one initially breaking the connection, however, we started experiencing different types of errors once that one was resolved. The change here is to attempt to restart the connection regardless of the error until Sarama improves their internal `sarama.ClientAdmin` connection handling.